watermarker/src/commonMain/kotlin/watermarks/TextWatermark.kt

```diff
@@ -17,37 +17,46 @@ import kotlin.jvm.JvmName
  * text as content.
  *
  * TextWatermark can be instantiated using the companion functions. The functions `new`, `raw`,
- * `compressed`, `sized`, and `compressedAndSized` allow to create a new TextWatermark from a String
- * and specifying the format of the produced Trendmark.
+ * `compressed`, `sized`, `checked`, `hashed`, `compressedAndSized`, `compressedAndChecked`,
+ * `compressedAndHashed`, `sizedAndChecked`, `sizedAndHashed`, `compressedSizedChecked` and
+ * `compressedSizedHashed` allows creation of a new TextWatermark from a String and specifying
+ * the format of the produced Trendmark.
+ *
+ * The function `fromTrendmark` allows parsing a supported Trendmark into a TextWatermark, giving
+ * direct access to the contained text without having to consider the format of the Trendmark.
  *
  * Sized TextWatermarks will create Trendmarks that encode the size of the Trendmark into the
  * watermark.
  *
+ * Checked TextWatermarks will create Trendmarks that encode a CRC32 checksum into the watermark.
+ *
+ * Hashed TextWatermarks will create Trendmarks that encode a SHA3256 hash into the watermark.
+ *
  * Compressed TextWatermarks will compress the Text using a compression algorithm. This can be
  * useful when the watermark text is very long, but it might reduce the watermark robustness.
  *
- * The function `fromTrendmark` allows to parse a supported Trendmark into a TextWatermark, giving
- * direct access to the contained text without having to consider the format of the Trendmark.
  */
 @JsExport
 class TextWatermark private constructor(
     var text: String,
     private var compressed: Boolean = false,
     private var sized: Boolean = false,
+    private var checked: Boolean = false,
+    private var hashed: Boolean = false,
```

Comment:
```suggestion
    private var crc32: Boolean = false,
    private var sha3256: Boolean = false,
```

Reply:
My decision to go with the names I did was mostly informed by hesitance to rename the existing "compressedAndSized" function and awkwardness of names like "compressedAndSHA3256". Preferably I'd like to drop the "And" from all companion functions in TextWatermark and use variable names "crc32" and "sha3256" as suggested.